An Untitled Mining Game/An Untitled Mining Game/scripts/scr_consume_item_from_inventory/scr_consume_item_from_inventory.gml

```diff
@@ -4,23 +4,34 @@ function scr_consume_item_from_inventory (item_number, item_amount) {
 with(obj_inventory) {
 		var ds_inv = ds_inventory
 		var item_found = false;
-		var ii = 0; repeat(inv_slots){ // look through inventory for pebble
-			if (ds_inv[# 0, ii] == item_number) {
-				if (ds_inv[# 1, ii] >= item_amount) {
-					ds_inv[# 1, ii] -= item_amount;
+		var amount_found = 0;
+		var ii = 0; repeat(inv_slots){ // look through inventory for item
+			if (ds_inv[# 0, ii] == item_number) { //we found some
+				//this stack doesn't have enough
+				if (ds_inv[# 1, ii] <= (item_amount - amount_found)) {
+					//add to amount
+					amount_found += ds_inv[# 1, ii];
+					//empty slot
+					ds_inv[# 0, ii] = item.none; // item = none
+					ds_inv[# 1, ii] = 0; // amount = none
+				}
```

Comment:
Ah, instances like this. This code seems to perform the same function as the one I commented on above, but is implemented in multiple locations.